### PR TITLE
Added validations to models

### PR DESCRIPTION
**Pull request by JupitersLB**
Added normal validations, although unsure how much we would want to ensure was present. 
Added scope to the subscription and BasketItem so that it ensures both have uniqueness when comparing basket_id/user_id and basket_id/product_id.

### CODE REVIEW

File: app/models/basket.rb [hunk 1]
```diff
@@ -2,4 +2,6 @@ class Basket < ApplicationRecord
   belongs_to :user
   has_many :subscriptions, dependent: :destroy
   has_many :basket_items, dependent: :destroy
+  validates :user_id, :price, :title, :availability, :description, presence: true
+  validates :title, :description, uniqueness: true
```
Comment (davide32123) [inline app/models/basket.rb:6]: Why should description also be unique?

Comment (JupitersLB) [reply app/models/basket.rb:6]: My concern would be a user creating 100s of the same basket. Perhaps, the uniqueness should be that a title/description pair should be unique.

Comment (JupitersLB) [reply app/models/basket.rb:6]: validates :title, uniqueness: { scope: :description }. Something like this be better?

Comment (davide32123) [reply app/models/basket.rb:6]: Yes I prefer this version, could you update?

**Pull request merged**